src/server/auth/server/roles.go

```diff
@@ -171,6 +171,40 @@ func init() {
 		},
 	})
 
+	// Project related roles
+	projectViewer := registerRole(&auth.Role{
+		Name:          auth.ProjectViewer,
+		ResourceTypes: []auth.ResourceType{auth.ResourceType_PROJECT, auth.ResourceType_CLUSTER},
+		Permissions: []auth.Permission{
+			auth.Permission_PROJECT_LIST_REPO,
+		},
+	})
+
+	projectWriter := registerRole(&auth.Role{
+		Name:          auth.ProjectWriter,
+		ResourceTypes: []auth.ResourceType{auth.ResourceType_PROJECT, auth.ResourceType_CLUSTER},
+		Permissions: combinePermissions(projectViewer.Permissions, []auth.Permission{
+			auth.Permission_PROJECT_LIST_REPO,
+		}),
+	})
+
+	projectOwner := registerRole(&auth.Role{
+		Name:          auth.ProjectOwner,
+		ResourceTypes: []auth.ResourceType{auth.ResourceType_PROJECT, auth.ResourceType_CLUSTER},
+		Permissions: combinePermissions(projectWriter.Permissions, []auth.Permission{
+			auth.Permission_PROJECT_DELETE,
+			auth.Permission_REPO_DELETE,
+		}),
+	})
+
+	projectCreator := registerRole(&auth.Role{
+		Name:          auth.ProjectCreator,
+		ResourceTypes: []auth.ResourceType{auth.ResourceType_CLUSTER, auth.ResourceType_CLUSTER},
```

Comment:
remove duplicate `auth.ResourceType_CLUSTER`

Reply:
thanks for the catch!